app/docs/0.10.x/proxy.md

```diff
@@ -0,0 +1,759 @@
+ones for routing incoming traffic are `hosts`, `uris` and `methods`.
+
+If Kong cannot determine to which API a given request should be routed to, it
+will response with:
+
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+Server: kong/<x.x.x>
+
+{
+    "message": "no API found with those values"
+}
+```
+
+### Reminder: how to add an API to Kong
+
+Before goign any further, let's take a few moments to make sure you know how to
+add an API to Kong.
+
+As explained in the [Adding your API][adding-your-api] quickstart guide, Kong is
+configured via its internal [Admin API][API] running by default on port `8001`. 
+Adding an API to Kong is as easy as sending an HTTP request:
+
+```bash
+$ curl -i -X POST http://localhost:8001/apis/ \
+    -d 'name=my-api' \
+    -d 'upstream_url=http://my-api.com' \
+    -d 'hosts=my-api.com' \
+    -d 'uris=/my-api' \
+    -d 'methods=GET,HEAD'
+HTTP/1.1 201 Created
+...
+```
+
+This request instructs Kong to register an API named "my-api", reachable at
+"http://my-api.com". It also specifies various routing properties. In reality,
+**only one of** `hosts`, `uris` and `methods` is actually required. More on this
+later.
+
+Adding such an API would mean that you configured Kong to proxy all incoming 
+requests matching `hosts`, `uris`, and `methods` to `http://my-api.com`.
+Kong being a transparent proxy, it will forward the request to your upstream
+service - practically - untouched (exception made of various headers such as
+`Connection`, or in case a plugin performs some transformations to it).
+
+[Back to TOC](#table-of-contents)
+
+### Routing capabilities
+
+Let's now investigate in details how Kong matches a request to the configured
+`hosts`, `uris` and `methods` properties of your API:
+
+- All of those fields are **optional**, but at least **one of them** must be 
+specified.
+- For a request to match an API, all of configured fields **must** be satisfied
+(present in the client HTTP request).
+- All of those fields accept one or more values. A field only needs one of 
+values to be considered a match.
+
+Let's go through a few concrete examples. Consider such an API:
+
+```json
+{
+    "name": "my-api",
+    "upstream_url": "http://my-api.com",
+    "hosts": ["example.com", "service.com"],
+    "uris": ["/foo", "bar"],
+    "methods": ["GET"]
+}
+```
+
+Some of the possible requests matching this API could be:
+
+```http
+GET /foo HTTP/1.1
+Host: example.com
+```
+
+```http
+GET /bar HTTP/1.1
+Host: service.com
+```
+
+```http
+GET /foo/hello/world HTTP/1.1
+Host: example.com
+```
+
+All three of those requests satisfy all of the three conditions set in the API
+definition.
+
+However, the following requests would **not** match this API:
+
+```http
+GET / HTTP/1.1
+Host: example.com
+```
+
+```http
+POST /foo HTTP/1.1
+Host: example.com
+```
+
+```http
+GET /foo HTTP/1.1
+Host: foo.com
+```
+
+All three of those requests are only satisfying two out of the three configured
```

Comment:
`those` should be `these`

`only satisfying` should be `satisfying only`

`out of the three` should be `of three`